src/Hamiltonians/ModifiedHamiltonian.jl

```diff
@@ -0,0 +1,148 @@
+"""
+    abstract type ModifiedHamiltonian{T} <: AbstractHamiltonian{T} end
+
+Abstract type for defining wrappers over [`AbstractHamiltonian`](@ref)s that modify diagonal
+and off-diagonal elements via the functions [`modify_diagonal`](@ref Main.Hamiltonians) and
+[`modify_offdiagonal`](@ref Main.Hamiltonians).
+
+The `ModifiedHamiltonian` can only be used to implement wrappers that modify the
+(off)-diagonals individually and cannot be used to introduce additional off-diagonal
+elements to the Hamiltonian.
+
+The following need to be implemented
+* [`parent_operator`](@ref Main.Hamiltonians)
+* [`modify_diagonal`](@ref Main.Hamiltonians)
+* [`modify_offdiagonal`](@ref Main.Hamiltonians)
```

Comment:
As it is right now, the links point to the docstring of the `Hamiltonians` module. Was that intended? I think it'll be better to link to the docstring of the respective function.

```suggestion
* [`parent_operator`](@ref)
* [`modify_diagonal`](@ref)
* [`modify_offdiagonal`](@ref)
```

Reply:
I always get confused how to link to unexported functions